src/physics/collision_scene.c

```diff
@@ -514,6 +514,12 @@ void collisionSceneGetPortalTransform(int fromPortal, struct Transform* out) {
     transformConcat(gCollisionScene.portalTransforms[1 - fromPortal], &inverseA, out);
 }
 
+void collisionSceneGetPortalRotation(int fromPortal, struct Quaternion* out) {
```

Comment:
> Since the heavier computations are only done once per grab/throw, can you continue to use `collisionSceneGetPortalTransform()` to reduce duplication?
> 
> Similarly, `playerPortalGrabTransform()` could just check if `grabPoint` is `NULL` and ignore it if so.
> 
> When performance is not at risk, I prefer to avoid functions that are mostly identical.

Using `playerPortalGrabTransform()` with a nullpointer check is a good idea. I don't see why we shouldn't have `collisionSceneGetPortalRotation()` though. Even without the optimizations in mind, it's just cleaner to be able to call this without having to set up a Transform struct that won't be used.

Reply:
I agree that it would be useful if/when we have code that requires it. With the `playerPortalGrabTransform()` `NULL` change though, `collisionSceneGetPortalRotation()` is now unused.

Reply:
`playerPortalGrabTransform()` can be rearranged, but such a change would be going quite far to avoid simply reusing `collisionSceneGetPortalTransform()`.

I agree that `collisionSceneGetPortalRotation()` is preferable in situations where an entire `Transform` isn't needed, but that's not really the case right now. If `grabPoint` is `NULL` this is technically true, but a `Transform` is already created by the function regardless. Avoiding it in this non-performance-critical code path is needlessly convoluted.

I view it as similar to only updating the player's forward direction if they pressed the corresponding button, rather than just updating all directions at once. There's a line we have to draw between optimizations and simplicity.

I'm all for adding it if something else ends up needing it, but prefer to avoid premature additions. This is one small instance, but I'm trying to be diligent.

Reply:
It's fine. I'm not really convinced by your reasoning, but I don't have to be. In the end it's also a pretty unimportant change. Thanks for reviewing and merging though.